src/system/NetworkNamespace.cpp

```diff
@@ -27,29 +27,23 @@ void returnNetworkNamespace(std::shared_ptr<NetworkNamespace> ns)
 
 std::shared_ptr<NetworkNamespace> claimNetworkNamespace()
 {
+    faabric::util::UniqueLock lock(namespacesLock);
     if (namespaces.empty() && !namespacesInitialised) {
-        faabric::util::UniqueLock lock(namespacesLock);
-
-        if (namespaces.empty() && !namespacesInitialised) {
-            namespacesInitialised = true;
-
-            // Note that the availability of namespaces depends on the Faasm
-            // configuration for the relevant host
-            // TODO - connect these two up properly
```

Comment:
afaict we are already taking the host config into account in the `NetworkNamespace` constructor.

Reply:
👍

Reply:
The host config is used in the `NetworkNamespace` constructor, but AFAICT it's just to get the name. What this TODO means, is that the number of network namespaces that are actually created on the host is not in any way connected with the `nNamespaces` variable we have here (which is arbitrarily set to `cores * 10`). This means that this class could allocate a namespace which doesn't actually exist on the host. 

So the TODO should stay, but perhaps worded more explicitly.

Reply:
:ok_hand: